BasiliskII/src/Unix/video_x.cpp

```diff
@@ -1794,6 +1808,11 @@ bool VideoInit(bool classic)
 	else if (default_height > DisplayHeight(x_display, screen))
 		default_height = DisplayHeight(x_display, screen);
 
+	// for classic Mac, make sure the display width is 8
+	if (classic) {
+		default_width = (default_width << 3) >> 3;
+	}
```

Comment:
Hmm, can you refactor this logic to a helper function that can be shared by all three files? Otherwise, duplicating this logic is pretty error prone.

Reply:
There are a large chunk of quite unique logic among SDL1, SDL2 and XWindows have  to initialize the width and the height.

Also, the add_mode functions in XWindows are different from SDL1, SDL2.